src/main/scala/com/mozilla/telemetry/pings/package.scala

```diff
@@ -0,0 +1,225 @@
+                   Type: Option[String],
+                   appBuildId: String,
+                   appName: String,
+                   appUpdateChannel: Option[String],
+                   appVendor: Option[String],
+                   appVersion: Double,
+                   clientId: Option[String],
+                   creationTimestamp: Option[Float],
+                   docType: Option[String],
+                   documentId: Option[String],
+                   geoCity: Option[String],
+                   geoCountry: String,
+                   normalizedChannel: String,
+                   os: String,
+                   sampleId: Option[Double],
+                   sourceName: Option[String],
+                   sourceVersion: Option[Int],
+                   submissionDate: String,
+                   telemetryEnabled: Option[Boolean],
+                   // Common fields preparsed by hindsight
+                   `environment.build`: Option[EnvironmentBuild],
+                   `environment.settings`: Option[Settings],
+                   `environment.system`: Option[System],
+                   `environment.profile`: Option[Profile],
+                   `environment.addons`: Option[Addons],
+                   // Main ping fields preparsed by hindsight
+                   `payload.simpleMeasurements`: JValue,
+                   `payload.keyedHistograms`: JValue,
+                   `payload.histograms`: JValue,
+                   `payload.info`: JValue
+                 )
+
+  case class CrashPayload(
+                           crashDate: String,
+                           processType: Option[String],
+                           hasCrashEnvironment: Option[Boolean],
+                           metadata: Option[Map[String, String]],
+                           version: Option[Int]
+                         )
+
+  case class CrashPing(
+                        application: Application,
+                        clientId: Option[String],
+                        payload: CrashPayload,
+                        // Environment is omitted it's partially available under meta
+                        meta: Meta
+                      ){
+
+    def isMainCrash(): Boolean = {
+      payload.processType.getOrElse("main") == "main"
+    }
+
+    def timestamp(): Timestamp = {
+      new Timestamp(this.meta.Timestamp / 1000000)
+    }
+  }
+
+  case class PayloadInfo(subsessionLength: Option[Int])
+
+  case class MainPing(
+                       application: Application,
+                       clientId: Option[String],
+                       // Environment omitted because it's mostly available under meta
+                       meta: Meta
+                     ){
+    def getCountHistogramValue(histogram_name: String): Int = {
+      try {
+        this.meta.`payload.histograms` \ histogram_name \ "values" \ "0" match {
+          case JInt(count) => count.toInt
+          case _ => 0
+        }
+      } catch { case _: Throwable => 0 }
```

Comment:
A missing histogram should not generate a value of 0 but something equivalent to "NA" (e.g. null), as the semantics is quite different. The same comment applies elsewhere in this class.